### PR TITLE
Implement pre/post hook support

### PR DESCRIPTION
**Pull request by sideshowdave7**
(Edited)
What:
Issue #9 

Implement an optional setup callable (or path to callable as a str) that will run immediately when the worker is spawned, configurable through the eventmq.conf file

Tests:

Added Unit tests for run_setup

### CODE REVIEW

**Comment by com4** [comment]
I'm not sure why this is needed. Can't the person writing the job just put the code that's needed to run before and after their job?

**Comment by sideshowdave7** [comment]
It's been requested a few times, I've put a couple use cases in https://github.com/eventmq/eventmq/issues/9

**Comment by sideshowdave7** [comment]
This would be for project level things, so a developer could create a job decorator with these hooks defined

**Comment by com4** [comment]
I read the use cases. It seems to needlessly complicate `defer_job`.  I can see a pre/post hook for creating the worker instance which would be the use case for setting up the django environment, but I don't think we need a pre/post hook for every job.

**Comment by sideshowdave7** [comment]
Yea I did consider just running it when a worker first gets instantiated.  Although the implementation would be just as complex for that I'm thinking.

**Comment by com4** [comment]
It doesn't complicate implementation for the user though. pre/post hooks make it harder to see what a job does for someone reading the code.  I think the job should explicitly put it's own setup/teardown code inside the job itself so it's easier for someone to debug.

something like `./emq-jobmanger --Q default,analytics --pre-hook django.setup` would better solve the use cases IMO. Then for each worker instance that's created it would execute the defined callable once, but not for every job.

**Comment by sideshowdave7** [comment]
Ah that makes sense, I would like the option to specify different hooks by queue as well so i like that approach

**Comment by com4** [comment]
nice. i like the idea of specifying a hook by queue.  It would be cool also if someone can define those pre/post hooks in the config so they can get replicated to other machines.

**Comment by sideshowdave7** [comment]
Currently I don't think we can easily do hooks per queue since all the workers on a box do jobs for all queues - we'd need to implement dedicated workers for each queue and I'm not sure thats worth the effort.

File: eventmq/worker.py [hunk 1]
```diff
@@ -148,5 +161,26 @@ def _run(payload):
     except Exception as e:
         logger.exception(e)
         return str(e)
+
     # Signal that we're done with this job
     return 'DONE'
+
+
+def run_setup(setup_path, setup_callable):
+    logger.debug("Running setup for worker id {}".format(os.getpid()))
```
Comment (synic) [inline eventmq/worker.py]: Might be nice to also log the `setup_path` and `setup_callable`

Comment (sideshowdave7) [inline eventmq/worker.py]: Added this as well as a version bump

**Pull request merged**